### PR TITLE
Follow-up non-verbose logs

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

Change how we invoke quiet.sh as per https://github.com/Mbed-TLS/mbedtls/pull/8678#discussion_r1513250581. Fixes https://github.com/Mbed-TLS/mbedtls/issues/8893

Also addresses an issue where the scripts would recurse forever if `PATH` was set to something like `$PATH:tests/scripts/quiet` (or presumably if using a symlink, I didn't test) as it would fail to match the path in `$0` against the output of `type -ap`.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required
- [x] **backport** #8895 
- [x] **tests** not required

### CODE REVIEW

**Comment by tom-cosgrove-arm** [comment]
As a reviewer I'd actually find this easier if it was rebased on top of development (I assume it's not as this is adding the scripts that the previous PR already added)

File: tests/scripts/quiet/quiet.sh [hunk 1]
```diff
@@ -22,9 +22,16 @@
...
+# Function to normalise paths
+get_real_filename() {
+    leafname="$(basename "$1")"
+    ( cd $(dirname "$1"); echo "$(pwd)/$leafname" )
+}
+
+# Normalise path to wrapper script
+WRAPPER_WITH_PATH=$(get_real_filename "$0")
+# Identify original tool (compare normalised path to WRAPPER_WITH_PATH to avoid recursively calling the same wrapper script)
+ORIGINAL_TOOL="$(for p in $(type -ap "$TOOL"); do get_real_filename "$p" ; done | grep -v -Fx "$WRAPPER_WITH_PATH" | head -n1)"
```
Comment (gilles-peskine-arm) [inline tests/scripts/quiet/quiet.sh:34]: This code is rather complicated. You can use the `-ef` test operator to check if two paths point to the same file (not POSIX, but present in bash as well as almost all shells you'll encounter these days), avoiding the complex file name manipulations and following file symlinks (which isn't really a concern here).

`for p in $(type -ap "$TOOL")` breaks with paths containing spaces. Parsing a newline-delimited list in POSIX sh is unfortunately fiddly. (Bash ≥4 has `mapfile` and zsh has `"${(@f)$(…)}"` but that doesn't help with macOS's `/bin/bash` which is 3.2.) One solution is to disable globbing and set `IFS` to newlines only. Another way is `IFS= read -r line`. Note that `somecommand | read … line; … $line` doesn't work because the right-hand side of the pipe operator runs in a subshell so the `line` variable is not set in the following command. But you can use `read … <(somecommand)` as long as you don't need to catch failures of `somecommand`.

```
while IFS= read -r ORIGINAL_TOOL < <(type -ap -- "$TOOL"); do
    if ! [[ $ORIGINAL_TOOL -ef $0 ]]; then break; fi
done
```

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]

File: tests/scripts/quiet/quiet.sh (hunk 1 above)
Comment (daverodgman) [reply tests/scripts/quiet/quiet.sh]: Thanks, this is cleaner. But AIUI the redirection needs to happen after the `done`?

Comment (gilles-peskine-arm) [reply tests/scripts/quiet/quiet.sh]: Oh, yes, sorry. The redirection where I put it only works in the common case where `type` only emits one line.

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM

File: tests/scripts/quiet/quiet.sh [hunk 2]
```diff
@@ -22,9 +22,13 @@
 #              be silenced, e.g. " --version | test ". In this example, "make lib test" will
 #              not be silent, but "make lib" will be.
 
-# Locate original tool
-TOOL_WITH_PATH=$(dirname "$0")/$TOOL
-ORIGINAL_TOOL=$(type -ap "${TOOL}" | grep -v -Fx "$TOOL_WITH_PATH" | head -n1)
+# Identify path to original tool. There is an edge-case here where the quiet wrapper is on the path via
+# a symlink or relative path, but "type -ap" yields the wrapper with it's normalised path. We use
```
Comment (tom-cosgrove-arm) [inline tests/scripts/quiet/quiet.sh:26]: (nit) It's `its` not `it's` for possessive

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM, with grammar fixed or not
